src/olympia/files/helpers.py

```diff
@@ -60,6 +64,32 @@ def file_tree(files, selected):
     return jinja2.Markup('\n'.join(output))
 
 
+def extract_file(viewer, **kw):
+    # This message is for end users so they'll see a nice error.
+    msg = Message('file-viewer:%s' % viewer)
+    msg.delete()
+    task_log.debug('Unzipping %s for file viewer.' % viewer)
+
+    try:
+        locked = viewer.extract()
+
+        if locked:
+            info_msg = _(
+                'File viewer is locked, extraction for %s could be '
+                'in progress. Please try again in approximately 5 minutes.'
+                % viewer)
+            msg.save(info_msg)
+    except Exception, err:
+        if settings.DEBUG:
+            msg.save(_('There was an error accessing file %s. %s.') %
```

Comment:
The error won't be in the translated string but yeah, we could potentially append it. Not sure if that makes a difference, I usually like having all arguments around in the translation too